### PR TITLE
Feature/templating-service

### PR DESCRIPTION
**Pull request by friedow**

### CODE REVIEW

File: platforms/testplatform/templates/global-announcement/MAIL.mustache [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+    <head>
+      <meta http-equiv='Content-Type' content='text/html; charset=utf-8'>
+      <title>openHPI</title>
+    </head>
+    <body>
+      <h1>{{ title }}</h1>
+      <span>{{ language.description }}</span>
+      <p>Greetings {{ user.name }}!</p>
+    </body>
+  </html>
```
Comment (hefowe) [inline platforms/testplatform/templates/global-announcement/MAIL.mustache]: Better use a more general example template (without Postman or openHPI).

File: src/services/TemplatingService.ts [hunk 2]
```diff
@@ -13,79 +16,86 @@ export default class TemplatingService {
...
 
-  private static insertMessagePayload(template: any, payload: any, platformId: string): any {
-    const config = Utils.getPlatformConfig(platformId);
-    return {
-      from: config.mail.defaults.from,
-      data: template.data,
-    };
+  private static parseMessageTemplate(template: Template) {
+    for (const key in template) {
+      // && typeof template[key] === 'string'
```
Comment (hefowe) [inline src/services/TemplatingService.ts]: Why do we have this comment?

File: package.json [hunk 3]
```diff
@@ -20,11 +20,15 @@
   },
   "homepage": "https://github.com/schul-cloud/node-notification-service",
   "dependencies": {
+    "@types/js-yaml": "^3.11.2",
```
Comment (hefowe) [inline package.json]: @friedow You changed the position of these types, right?

File: src/utils.ts [hunk 4]
```diff
@@ -1,9 +1,17 @@
...
-    }
+  public static getPlatformConfig(platformId: string): any {
+    return require(`../platforms/${platformId}/config.json`);
+  }
 
-    public static getTemplate(platformId: string, templateId: string, type: string): any {
-        return require(`../platforms/${platformId}/templates/${templateId}/${type}.json`);
-    }
+  public static loadTemplate(platformId: string, templateId: string, type: string): Template {
+    const templatePath = path.join(__dirname, '..', 'platforms', platformId, 'templates', templateId, `${type}.mustache`);
```
Comment (hefowe) [inline src/utils.ts]: `WARNING: --/utils.ts[12, 1]: Exceeds maximum line length of 120`
Can you fix this?

File: src/services/TemplatingService.ts [hunk 5]
```diff
@@ -13,79 +16,85 @@ export default class TemplatingService {
...
 
   public createPushMessage(user: UserRessource, device: string): firebaseMessaging.Message {
-    const template = this.getTemplateForUser(user, PUSH_MESSAGE);
+
+    const template = this.getTemplate(MAIL_MESSAGE);
+    const payload = this.getUserPayload(user);
+    const renderedTemplate = TemplatingService.renderMessageTemplate(template, payload);
     const push = {
       token: device,
       data: template.data,
```
Comment (hefowe) [inline src/services/TemplatingService.ts]: `template` should be `renderedTemplate `.

File: src/services/TemplatingService.ts [hunk 6]
```diff
@@ -13,79 +16,85 @@ export default class TemplatingService {
...
+    });
+  }
 
-  private static insertMessagePayload(template: any, payload: any, platformId: string): any {
-    const config = Utils.getPlatformConfig(platformId);
-    return {
-      from: config.mail.defaults.from,
-      data: template.data,
-    };
+  private static parseMessageTemplate(template: Template) {
```
Comment (hefowe) [inline src/services/TemplatingService.ts]: The templates for push messages are nested. This is currently not supported, right?

The PUSH template could look like this:
```
apns:
    payload:
        aps:
            title: "{{ message.title }}"
            body: "{{ message.body }}"
```

File: src/services/TemplatingService.ts [hunk 7]
```diff
@@ -13,140 +16,124 @@ export default class TemplatingService {
...
+      text: renderedTemplate.text,
+      html: renderedTemplate.html,
     };
     return mail;
   }
 
   public createPushMessage(user: UserRessource, device: string): firebaseMessaging.Message {
-    const template = this.getTemplateForUser(user, PUSH_MESSAGE);
+
+    const template = this.getTemplate(MAIL_MESSAGE);
```
Comment (hefowe) [inline src/services/TemplatingService.ts]: This should be `PUSH_MESSAGE`.

File: src/services/TemplatingService.ts (hunk 6 above)
Comment (friedow) [reply src/services/TemplatingService.ts]: Done

**Pull request merged**